lib/executable.dart

```diff
@@ -197,20 +192,15 @@ Future<void> main(List<String> args) async {
             processManager: globals.processManager,
           ),
       TemplateRenderer: () => const MustacheTemplateRenderer(),
-      TizenBuilder: () => TizenBuilder(
-            logger: globals.logger,
-            processManager: globals.processManager,
-            fileSystem: globals.fs,
-            artifacts: globals.artifacts,
-            usage: globals.flutterUsage,
-            platform: globals.platform,
-          ),
+      TizenBuilder: () => TizenBuilder(),
       TizenSdk: () => TizenSdk.locateSdk(),
       TizenValidator: () => TizenValidator(
             tizenSdk: tizenSdk,
             dotnetCli: dotnetCli,
+            fileSystem: globals.fs,
```

Comment:
Can we remove these global arguments also like `TizenBuilder`? What are the criteria for removing the global arguments?

Reply:
Good question. Explicitly feeding arguments to a class (thus not using globals) generally improves testability and clarity of the code (https://github.com/flutter/flutter/issues/47161), but that's not always the case and sometimes it's hard to say which way is better. My personal criteria is to use globals when either:

- Too many of them are used in the class so using globals makes the implementation convenient and cleaner.
- It's hard to avoid context injection in the test code or there's no plan to add tests for the class.